### PR TITLE
Misc updates

### PR DESCRIPTION
**Pull request by ricardoV94**
* Remove uses of tag.ignore_logprob which is no longer recognized by Aeppl, and would result in a UserWarning
* Retrieve RNG updates for non RandomVariables at compile time

### CODE REVIEW

File: pymc/distributions/logprob.py [hunk 1]
```diff
@@ -221,7 +222,11 @@ def joint_logpt(
 
     transform_opt = TransformValuesOpt(transform_map)
     temp_logp_var_dict = factorized_joint_logprob(
-        tmp_rvs_to_values, extra_rewrites=transform_opt, use_jacobian=jacobian, **kwargs
+        tmp_rvs_to_values,
+        extra_rewrites=transform_opt,
+        use_jacobian=jacobian,
+        warn_missing_rvs=False,
```
Comment (ricardoV94) [inline pymc/distributions/logprob.py:228]: Instead of warning, it will fail which is a more strict behavior

**Comment by ricardoV94** [comment]
This test keeps failing: test_sample_posterior_predictive_after_set_data_with_coords across PRs

**Review by twiecki — APPROVED** [review]
Skimmed it.

**Comment by ricardoV94** [comment]
Had missed one warning for the dist used in the GaussianRandomWalk

**Pull request merged**